modules/analytics/src/views/full/index.tsx

```diff
@@ -218,29 +216,41 @@ const Analytics: FC<any> = ({ bp }) => {
   }
 
   const fetchQnaQuestions = async () => {
-    const qnaIds = orderMetrics(getMetric('msg_sent_qna_count'))
-      .filter(m => m.name)
-      .slice(0, 10)
-      .map(m => m.name)
+    const metrics = orderMetrics(getMetric('msg_sent_qna_count').filter(metric => metric.subMetric)).slice(0, 10)
 
-    const fetchedQuestions = await Promise.all(
-      qnaIds.filter(id => !(id in qnaQuestionsCache)).map(id => fetchQnaQuestion(id.replace('__qna__', '')))
-    )
+    const topQnaQuestions = await Promise.all(
+      metrics.map(async ({ name: id, count }) => {
+        if (id in qnaQuestionsCache.found) {
+          return { count, id, question: qnaQuestionsCache.found[id] }
+        }
 
-    for (const {
-      data: { questions },
-      id
-    } of fetchedQuestions) {
-      const question = questions[lang.getLocale()] || questions[lang.defaultLocale] || Object.values(questions)[0]
-      qnaQuestionsCache[`__qna__${id}`] = question[0]
-    }
+        if (qnaQuestionsCache.notFound.has(id)) {
+          return { count, id }
+        }
 
-    const qnaQuestions = qnaIds.reduce((acc, id) => {
-      acc[id] = qnaQuestionsCache[id]
-      return acc
-    }, {})
+        let response
+        try {
+          response = await fetchQnaQuestion(id.replace('__qna__', ''))
+        } catch (e) {
+          qnaQuestionsCache.notFound.add(id)
+          return { count, id }
+        }
 
-    dispatch({ type: 'receivedQnaQuestions', data: { qnaQuestions } })
+        const {
+          data: { questions }
+        } = response
+        const question = (questions[lang.getLocale()] ||
```

Comment:
Should we display  the qna in the language currently being edited instead of users locale ?

If yes, you receive it in props as `contentLang` as this is the top level component

Reply:
I think the user's locale is better, since analytics does not relate to the current language being edited.